src/cudadecoder/cuda-decoder.cc

```diff
@@ -1585,7 +1594,11 @@ void CudaDecoder::ConcurrentGetRawLatticeSingleChannel(const ChannelId ichannel,
       "You need to call PrepareForGetRawLattice before "
       "ConcurrentGetRawLatticeSingleChannel" &&
       best_cost_idx >= 0);
+  fst_out->DeleteStates();  // reset out lattice
   const int32 nframes = NumFramesDecoded(ichannel);
+  // If no frames were decoded, leaving the lattice empty and return
+  if (nframes == 0) return;
+
```

Comment:
Moving the assert below the `nframes == 0` check allow it to run properly.

Reply:
Great, thanks